src/rgw/rgw_client_io.cc

```diff
@@ -13,8 +13,12 @@
 namespace rgw {
 namespace io {
 
-void BasicClient::init(CephContext *cct) {
-  init_env(cct);
+[[nodiscard]] int BasicClient::init(CephContext *cct) {
+  int init_error=0;
+  init_error = init_env(cct);
```

Comment:
can squash these two lines

Reply:
also, maybe the `[[nodiscard]]` should be at the class definition?

Reply:
in this case introducing [[nodiscard]] at class definition generates a warning as RGWLibIO which derives from the basicClient will not respect the nodiscard flag